cpp/src/arrow/util/bit_stream_utils.h

```diff
@@ -108,34 +108,39 @@ class BitWriter {
   int bit_offset_;   // Offset in buffered_values_
 };
 
+namespace detail {
+
+inline uint64_t ReadLittleEndianWord(const uint8_t* buffer, int bytes_remaining) {
```

Comment:
Will this have issue if `bytes_remaining` is 0?

Reply:
Probably not, as long as `buffer` is non-null.
Note that this is merely a small refactor of `BitReader` but may not be necessary to fix this issue.